### PR TITLE
MEN-4562: Extend test_update_control to verify deployment maps clean-up

### PR DESCRIPTION
**Pull request by lluiscampos**
Implementation at https://github.com/mendersoftware/mender/pull/790
Already approved at https://github.com/mendersoftware/meta-mender/pull/1385

### CODE REVIEW

**Comment by lluiscampos** [comment]
Testing together with https://github.com/mendersoftware/mender/pull/790 @ [Pipeline 318490283](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/318490283)

**Comment by lluiscampos** [comment]
New build at [Pipeline 319352823](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/319352823)

**Comment by lluiscampos** [comment]
The original design of these tests conflicts with [these lines](https://github.com/mendersoftware/mender/blob/a63d67c38e349fbf81bbc14c40b8858e1daef843/app/mender.go#L366-L367) from the client, as the scenario was setting Control Maps via D-Bus with IDs matching the future deployment.

I need to redesign the tests to send the maps via mock server. I'll work on that next week.

**Comment by lluiscampos** [comment]
> I need to redesign the tests to send the maps via mock server. I'll work on that next week.

I redesigned the test to send the "pause" control map together with the update from the server, and then override it via DBus API.

**Comment by lluiscampos** [comment]
New build at [Pipeline 319815091](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/319815091)

File: tests/acceptance/test_update_control.py [hunk 1]
```diff
@@ -443,19 +453,125 @@ def make_and_deploy_artifact():
...
+            # update; then once the client is paused, the map is overridden via DBus API
+            make_and_deploy_artifact(
+                connection,
+                bitbake_variables["MENDER_DEVICE_TYPE"],
+                deployment_id=MUID,
+                update_control_map=case["pause_map"],
+            )
+            wait_for_state(case["pause_state"])
+            set_update_control_map(connection, case["continue_map"])
+            wait_for_state("Cleanup")
```
Comment (kacf) [inline tests/acceptance/test_update_control.py:564]: As a safeguard to make sure the map is actually working (and not just being silently ignored or simply not picked up), maybe check the status of the first deployment here, either success or failure depending on the test case?

**Review by kacf — APPROVED** [review]
One small improvement suggestion below, otherwise looks good!

File: tests/acceptance/test_update_control.py (hunk 1 above)
Comment (lluiscampos) [reply tests/acceptance/test_update_control.py:564]: Okey, fair enough.

**Pull request merged**